### PR TITLE
Selectable card icon

### PR DESCRIPTION
**Pull request by bh247484**
Added the following kits:  **Selectable Card Icon** and **Selectable Icon**

Things worth reviewing...

- JSX Conditional Rendering in Selectable Card Icon kit.  Bit messy, maybe there's a better way to do this?
- Sass structure  / organization
- Make sure I implemented classnames, aria, and data prop passing correctly
- Form input ids are numbers.  Is that okay?
- Single Select option props (multi, name, etc).  Am I implementing them correctly?  Pretty much just replicated 'Selectable Card' functionality.

Final Note:
My bundle gem is broken after rebasing and pushing changes.  Probably local issue not related to this PR but thought I should mention it here just in case.

### CODE REVIEW

File: app/pb_kits/playbook/pb_selectable_card_icon/_selectable_card_icon.scss [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        
+        &:not(:checked) ~ label {
+            color: $slate;
+            transition-duration: $transition_short;
+            transition-timing-function: $easeOut;
+        }
+    }
+
+    &[class*=_disabled] {
+        opacity: 0.55;
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_selectable_card_icon/_selectable_card_icon.scss:31]:
```suggestion
        opacity: $opacity_5;
```

File: app/pb_kits/playbook/pb_selectable_card_icon/_selectable_card_icon.scss [hunk 2]
```diff
@@ -0,0 +1,65 @@
+@import "../tokens/colors";
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_selectable_card_icon/_selectable_card_icon.scss:1]: In general, this seems to have more spacing than other files. Do you have 4 spaces for tabs? I believe 2 is what most folks have

Comment (bh247484) [reply app/pb_kits/playbook/pb_selectable_card_icon/_selectable_card_icon.scss:1]: Changed VS Code settings to 2 spaces.  Cleaned up Sass files indentation.  Will clean up rails/ruby as well.

File: app/pb_kits/playbook/pb_selectable_icon/_selectable_icon.scss [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+		transition-property: color;
+		transition-duration: $transition_short;
+		transition-timing-function: $easeIn;
+
+		&:checked ~ label * {
+			color: $primary;
+		}
+		
+		&:not(:checked) ~ label {
+			color: $slate;
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_selectable_icon/_selectable_icon.scss:46]:
```suggestion
			color: $text_lt_lighter;
```

File: app/pb_kits/playbook/pb_selectable_icon/_selectable_icon.scss [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+			&:checked ~ label * {
+				color: $active_dark !important;
+			}
+				
+			&:not(:checked) ~ label {
+				
+				svg {
+					color: $slate;
+				}
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_selectable_icon/_selectable_icon.scss:76]:
```suggestion
					color: $text_dk_lighter;
```

Comment (jasoncypret) [reply app/pb_kits/playbook/pb_selectable_icon/_selectable_icon.scss:76]: Don't target the SVG. FA will color the icon based on the text properties you add to the parent

**Comment by thestephenmarshall** [comment]
@bh247484 it appears as though this one needs to be rebased as I'm seeing commits that shouldn't be showing up.

**Comment by thestephenmarshall** [comment]
I spun this one up and it is working perfectly. @bh247484 let's go over the rebase in the morning.

File: app/pb_kits/playbook/pb_selectable_icon/_selectable_icon.html.erb [hunk 5]
```diff
@@ -0,0 +1,24 @@
+<%= content_tag(:div,
+    id: object.id,
+    data: object.data,
+    class: object.classname) do %>
+
+    <% if object.inputless %>
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_selectable_icon/_selectable_icon.html.erb:6]: Let's discuss at dailies but I'd like to rethink this prop.

File: app/pb_kits/playbook/pb_selectable_icon/_selectable_icon.jsx [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+
+  const inputIdPresent = inputId !== null ? inputId : name
+
+  return (
+    <div
+        {...ariaProps}
+        {...dataProps}
+        className={classnames(css, className)}
+    >
+      { inputless ? (
```
Comment (drborges) [inline app/pb_kits/playbook/pb_selectable_icon/_selectable_icon.jsx:66]: I think the `<If condition={inputless}>` from [jsx-control-statements](https://github.com/AlexGilleran/jsx-control-statements#if) may help with improving code readability here.

Comment (bh247484) [reply app/pb_kits/playbook/pb_selectable_icon/_selectable_icon.jsx]: Ah, that's great.  Had seen this used elsewhere, didn't know we had access to this in our codebase.  Much cleaner than a long stringy ternary.

Comment (bh247484) [reply app/pb_kits/playbook/pb_selectable_icon/_selectable_icon.jsx]: Am seeing now it's just a babel feature... who knew?  Thanks for pointing this out!

**Pull request merged**